subprojects/docs/src/snippets/kotlinDsl/assignment/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,26 @@
+plugins {
+    `base`
+}
+
+// tag::assignment[]
+abstract class MyTask : DefaultTask() {
+    @get:Input
+    abstract val input: Property<String>
+    @get:OutputFile
+    abstract val output: RegularFileProperty
+
+    @TaskAction
+    fun execute() {
+        output.get().asFile.writeText(input.get())
+    }
+}
+
+tasks.register<MyTask>("myTask") {
+    input.set("Hello Property") // <1>
+    output.set(file("build/myTask/output.txt"))
+    input = "Hello Property" // <2>
+    ouput = file("build/myTask/output.txt")
+    input = provider { "Hello Property" } // <3>
+    output = provider { file("build/myTask/output.txt") }
```

Comment:
:lipstick: It would be better to demonstrate wiring from another property/provider instead of using `project.provider {}`.

Reply:
I changed the example, you can see it [here](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/62316250:id/distributions/gradle-8.1-docs.zip!/gradle-8.1-20230228190644%2B0000/docs/userguide/kotlin_dsl.html#kotdsl:assignment).

Is that any better, or it makes it more confusing since `<1>` and `<2>` are used in one block and `<3>` is used in another, WDYT?

Reply:
It is better. But I think we can simplify it further.
What about having all the assignments on the task, and none on the extension.
The snippet code could still have some values set on the extension but I would hide it from the manual.